hgtv/views/video.py

```diff
@@ -190,13 +189,31 @@ def make_presentz_json(video, json_value):
         unique_value = get_slideshare_unique_value(video.slides_url)
         d['chapters'][0]['slides'] = [{'time': str(key), "public_url": urllib.quote(video.slides_url), "url": 'https://slideshare.net/' + unique_value + "#" + str(val)} for key, val in json_value.items()]
     elif video.slides_source == u'speakerdeck':
-        #json to supply for presentz syncing
+        # json to supply for presentz syncing
         d['chapters'][0]['slides'] = [{'time': str(key), "url": 'https://speakerdeck.com/' + urllib.quote(video.slides_sourceid) + "#" + str(val)} for key, val in json_value.items()]
     return json.dumps(d)
 
 
-def add_new_video(channel, playlist):
+@app.route('/<channel>/<playlist>/new', methods=['GET', 'POST'])
+@lastuser.requires_login
+@render_with({'text/html': 'index.html.jinja2'}, json=True)
+@load_models(
+    (Channel, {'name': 'channel'}, 'channel'),
+    (Playlist, {'name': 'playlist', 'channel': 'channel'}, 'playlist'),
+    permission='new-video')
+def video_new(channel, playlist):
+    """
+    Add a new video
+    """
     form = VideoAddForm()
+    if request.method == 'GET':
+        if playlist is None:
+            cancel_url = channel.url_for()
+        else:
+            cancel_url = playlist.url_for()
+        html_form = render_form(form=form, title=u"New Video", submit=u"Add",
+                           cancel_url=cancel_url, ajax=False, with_chrome=False)
+        return {'channel': dict(channel.current_access()), 'playlist': dict(playlist.current_access()), 'form': html_form}
     if form.validate_on_submit():
         stream_playlist = channel.playlist_for_stream(create=True)
         video = Video(playlist=playlist if playlist is not None else stream_playlist)
```

Comment:
Replace these two lines with:

```python
video = Video(playlist=playlist if playlist is not None else channel.playlist_for_stream(create=True))
```

Reply:
`stream_playlist` is used in 2 more places below this piece of code and needed there.